internal/provider/resource_gpg_key.go

```diff
@@ -0,0 +1,279 @@
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"resource_path": schema.StringAttribute{
+				MarkdownDescription: "Path of the parent group.",
+				Description:         "Path of the parent group.",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+// Configure lets the provider implement the ResourceWithConfigure interface.
+func (t *gpgKeyResource) Configure(_ context.Context,
+	req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	t.client = req.ProviderData.(*tharsis.Client)
+}
+
+func (t *gpgKeyResource) Create(ctx context.Context,
+	req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	// Retrieve values from GPG key.
+	var gpgKey GPGKeyModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &gpgKey)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Create the GPG key.
+	created, err := t.client.GPGKey.CreateGPGKey(ctx,
+		&ttypes.CreateGPGKeyInput{
+			ASCIIArmor: gpgKey.ASCIIArmor.ValueString(),
+			GroupPath:  gpgKey.GroupPath.ValueString(),
+		})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating GPG key",
+			err.Error(),
+		)
+		return
+	}
+
+	// Map the response body to the schema and update the plan with the computed attribute values.
+	t.copyGPGKey(*created, &gpgKey)
+
+	// Set the response state to the fully-populated plan, whether or not there is an error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, gpgKey)...)
+}
+
+func (t *gpgKeyResource) Read(ctx context.Context,
+	req resource.ReadRequest, resp *resource.ReadResponse) {
+
+	// Get the current state.
+	var state GPGKeyModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get the GPG key from Tharsis.
+	found, err := t.client.GPGKey.GetGPGKey(ctx, &ttypes.GetGPGKeyInput{
+		ID: state.ID.ValueString(),
+	})
+	if err != nil {
+		if tharsis.IsNotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"Error reading GPG key",
+			err.Error(),
+		)
+		return
+	}
+
+	// Copy the from-Tharsis struct to the state.
+	t.copyGPGKey(*found, &state)
+
+	// Set the refreshed state, whether or not there is an error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (t *gpgKeyResource) Update(ctx context.Context,
+	req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	// Retrieve values from plan.
+	var plan GPGKeyModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var state GPGKeyModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !reflect.DeepEqual(plan, state) {
+		resp.Diagnostics.AddError(
```

Comment:
Doesn't seem like this should be needed now that we're using the RequiresReplace modifier.